frontend/src/pages/PipelineDetails.tsx

```diff
@@ -235,24 +239,49 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
     if (origin) {
       const msgRunOrRecurringRun = origin.isRecurring ? 'recurring run' : 'run';
       try {
+        // TODO(jlyaoyuli): change to v2 API after v1 is deprecated
         if (origin.isRecurring) {
-          origin.recurringRun = await Apis.jobServiceApi.getJob(origin.recurringRunId!);
+          origin.v1RecurringRun = await Apis.jobServiceApi.getJob(origin.recurringRunId!);
+          origin.v2RecurringRun = await Apis.recurringRunServiceApi.getRecurringRun(
+            origin.recurringRunId!,
```

Comment:
Won't one of these API call fail as always? There should be only V1 or V2 recurring runs, and the API call with different version should fail.

Reply:
No, both of the API call won't fail. The pipeline details here is like a router, so we should make both v1 and v2 API call and determine which version the run is, and then get the template string via either `pipeline_spec` or `pipeline_version_id`